annotation/tests/test_jobs.py

```diff
@@ -276,3 +279,100 @@ def test_get_tasks_to_delete(tasks: Tuple[ManualAnnotationTask]):
         [tasks[0], tasks[1], tasks[2], tasks[3], tasks[4], tasks[5]]
     )
     assert result == expected_result
+
+
+def test_get_jobs_by_files():
+    mock_session = MagicMock()
+    mock_session.query().join().order_by().all.return_value = (
+        (1, 1, JobStatusEnumSchema.finished),
+        (1, 2, JobStatusEnumSchema.pending),
+        (2, 3, JobStatusEnumSchema.in_progress),
+    )
+    expected_result = {
+        1: [
+            {
+                "id": 1,
+                "name": "job_name_1",
+                "status": JobStatusEnumSchema.finished,
+            },
+            {
+                "id": 2,
+                "name": "job_name_2",
+                "status": JobStatusEnumSchema.pending,
+            },
+        ],
+        2: [
+            {"id": 3, "name": None, "status": JobStatusEnumSchema.in_progress}
+        ],
+    }
+    with patch(
+        "annotation.jobs.services.collect_job_names",
+        return_value={1: "job_name_1", 2: "job_name_2"},
+    ):
+        result = get_jobs_by_files(mock_session, {1, 2, 3}, "test", "test")
+        assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    ("job_id", "attributes", "expected_result"),
+    (
+        (
+            1,
+            (Job,),
+            Job(
+                job_id=1,
+                name="test1",
+                callback_url="http://www.test.com",
+                annotators=[User(user_id="1")],
+                validation_type=ValidationSchema.cross,
+                is_auto_distribution=False,
+                categories=[categories],
+                deadline="2021-10-19T01:01:01",
+                tenant="test",
+            ),
+        ),
+        (
+            2,
+            (Job.name, Job.status),
+            ("job_name_2", JobStatusEnumSchema.finished),
+        ),
+        (3, (Job.name,), None),
+    ),
+)
+def test_get_job_attributes_for_post(
+    job_id: int,
+    attributes: Union[Tuple[Base], Tuple[InstrumentedAttribute, ...]],
```

Comment:
Could you clarify Tuple[Base]? I cannot really see that anywhere

Reply:
It means the Base class for Job model, but now that the code is refactored there is no need for that